### PR TITLE
feat(typescript): infer field names and type from `FormData`

### PR DESCRIPTION
**Pull request by iamandrewluca**

### CODE REVIEW

**Comment by erikras** [comment]
This pull request **introduces 1 alert** when merging b0c9db57a0fab6ffa4d4ca3d11505ad1c462ae61 into 0d3a88600d723976644e3df9073f329b8c513b3d - [view on LGTM.com](https://lgtm.com/projects/g/final-form/final-form/rev/pr-4aefb17342ed19366c18f150d9e8df7046600c47)

**new alerts:**

* 1 for Unused variable, import, function or class

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by erikras** [comment]
Ooh, that's cool.

**Comment by erikras** [comment]
Seems like it would not work for deep keys. e.g. `'address.street'`.

**Comment by iamandrewluca** [comment]
Hmm, I think, in this case typescript can't help us. Didn't thought of this.

File: src/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 export type Subscription = { [key: string]: boolean }
 export type Subscriber<V> = (value: V) => void
 export type IsEqual = (a: any, b: any) => boolean
+export type Partial<T> = { [P in keyof T]?: T[P] }
```
Comment (Andarist) [inline src/index.d.ts]: Partial is a builtin type, no need to create your own copy 😉

**Pull request closed without merging**